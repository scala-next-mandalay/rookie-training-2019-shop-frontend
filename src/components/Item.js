import React from 'react'
import PropTypes from 'prop-types'
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActions from '@material-ui/core/CardActions';
import Box from '@material-ui/core/Box'
import { BASEURL_ITEM_IMAGES } from '../constants'
import Button from '@material-ui/core/Button';
import AddShoppingCartIcon from '@material-ui/icons/AddShoppingCart';

const useStyles = makeStyles(theme => ({
  card: {
    maxWidth: 600,
    margin: theme.spacing(1),
  },
  media: {
    height: 0,
    //paddingTop: '56.25%', // 16:9
    paddingTop: '128%',
  },
  rightIcon: {
    marginLeft: theme.spacing(1),
  },
  cardContentArea: {
    marginTop: theme.spacing(1),
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
  },
  cardActionArea: {
    marginLeft: theme.spacing(2),
    marginRight: theme.spacing(2),
    marginBottom: theme.spacing(2),
  },
}));

const Item = ({ addCartItem, row }) => {
  const classes = useStyles()
  return (
    <Card className={classes.card}>
      <CardMedia
        className={classes.media}
        image={BASEURL_ITEM_IMAGES+row.image}
        //image={"http://127.0.0.1/dummyImage.jpg"}
        title={row.name}
      />
      <CardContent className={classes.cardContentArea}>
        <Box fontWeight={600}>
          {row.name}
        </Box>
        <Box>
          Price: {row.price} Ks
        </Box>
      </CardContent>
      <CardActions className={classes.cardActionArea}>
        <Button 
          variant="outlined" 
          fullWidth 
          color="primary"
          onClick={()=>{
            addCartItem(row)
          }}
        >
          Add to Cart
          <AddShoppingCartIcon className={classes.rightIcon} />
        </Button>
      </CardActions>
    </Card>
  )
}

Item.propTypes = {
  row: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    image: PropTypes.string,
  }).isRequired,
  addCartItem: PropTypes.func,
}

export default Item
