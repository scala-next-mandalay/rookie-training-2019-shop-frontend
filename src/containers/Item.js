import { connect } from 'react-redux';
import Item from '../components/Item';
import { addCartItem} from '../modules/cart';

const mapDispatchToProps = dispatch => ({
  addCartItem: (itemId) => dispatch(addCartItem(itemId)),
});

export default connect(
  null,
  mapDispatchToProps
)(Item);
