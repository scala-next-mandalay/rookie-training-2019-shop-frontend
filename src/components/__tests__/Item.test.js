import React from 'react'
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Parent from '../Parent';
import { act } from 'react-dom/test-utils';
import Item from '../Item';
/*global expect*/
/*global jest*/
/*global Event*/

let container;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.outerHTML = '<body></body>';
  container = null;
});

const row = [{ id: 1, name: "ver" ,price:4000 ,image: 'pic1' },
            { id: 2, name: "mer" ,price:3000 ,image: 'pic2' }];

let item=null;
const addCartItem = jest.fn((value) => {
  item = value;
});
describe("Item component", () => {
  it('matches the snapshot', () => {
    const ItemSnapshot = renderer.create(
      <Parent><Item 
      addCartItem={addCartItem} 
      row={row}
     /></Parent>
    ).toJSON();
    expect(ItemSnapshot).toMatchSnapshot();
  });
});

describe("Testing Item  Component", () => {
 it('testing Item add to cart button', () => {
    act(() => {
      ReactDOM.render((
        
        <Parent>
         <Item 
          addCartItem={addCartItem} 
          row={row[1]}/>
        </Parent>
  
      ), container);
   
    });
    const buttonArr=document.querySelectorAll('BUTTON');
    act(()=>{
       buttonArr[0].dispatchEvent(new Event('click', { bubbles: true }));
    });
    const expectedInfo={ id: 2, name: "mer" ,price:3000 ,image: 'pic2' };
    expect(addCartItem).toHaveBeenCalled();
    expect(item).toEqual(expectedInfo);
          
  
    });
});