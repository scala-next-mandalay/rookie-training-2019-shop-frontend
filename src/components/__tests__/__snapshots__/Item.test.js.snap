// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item component matches the snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
>
  <div
    className="MuiCardMedia-root makeStyles-media-2"
    style={
      Object {
        "backgroundImage": "url(\\"http://dg9c95ld8mycd.cloudfront.net/items/undefined\\")",
      }
    }
  />
  <div
    className="MuiCardContent-root makeStyles-cardContent-3"
  >
    <div
      className="MuiBox-root MuiBox-root-37 nameTxt"
    >
      Name 
    </div>
    <div
      className="MuiBox-root MuiBox-root-38 priceTxt"
    >
      <span>
        Price
      </span>
      :Price
       Ks
    </div>
  </div>
  <div
    className="MuiCardActions-root makeStyles-cardActions-4 MuiCardActions-spacing"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <div
          className="MuiBox-root MuiBox-root-61"
        >
          <span>
            Add Button
          </span>
        </div>
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0zm18.31 6l-2.76 5z"
            fill="none"
          />
          <path
            d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
